### PR TITLE
Only have the inline style if the activator slot is used

### LINKED ISSUE #3645: [Bug Report] "display: inline-block" CSS property on <v-menu> component
## Versions and Environment
**Vuetify:** 1.0.1
**Vue:** 2.5.16
**Browsers:** Chrome 65.0.3325.162
**OS:** Windows 10

### Steps to reproduce
Un-comment the `v-menu` component in the codepen to see how it affects the element below it

### Expected Behavior
Is there a particular reason why an invisible component needs `display: inline-block;` which affects visible components on the page?

### Actual Behavior
`<v-menu>` component has `display: inline-block;` which moves all the elements below it when you add it on the page.

### Reproduction Link
[https://codepen.io/anon/pen/WzjoGz](https://codepen.io/anon/pen/WzjoGz)

### PR DESCRIPTION
**Pull request by ebisbe**
## Description

Fixes #3645 

## Markup:

<details>

```vue
<template>
  <v-app>
    <div>
      <h2>Uncomment v-menu to see how it affects the tag below it</h2>
         <v-menu
          offset-y
          v-model="showMenu"
          absolute
          :close-on-content-click="false"
          :position-x="x"
          :position-y="y"
          :nudge-width="200"
        >
          <div>test</div>
        </v-menu>
      <p>this text will be moved down</p>
    </div>
  </v-app>
</template>

<script>
  export default {
    data: () => ({
    })
  }
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: src/components/VMenu/VMenu.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ export default {
     const data = {
       staticClass: 'v-menu',
       style: {
-        display: this.fullWidth ? 'block' : 'inline-block'
+        display: !this.fullWidth && this.$slots.activator ? 'inline-block' : 'block'
```
Comment (KaelWD) [inline src/components/VMenu/VMenu.js]: Don't set style at all if there's no activator

**Comment by johnleider** [comment]
I think this can go to master unless @KaelWD disagrees

**Comment by ebisbe** [comment]
I was not sure how to code it to be more accurate to your Programming style so suggestions are welcome.

**Comment by johnleider** [comment]
Code looks great, no worries there.

**Comment by KaelWD** [comment]
Oh yeah didn't catch that. 

@ebisbe Rebase onto master please.

**Comment by ebisbe** [comment]
`git rebase --onto master HEAD~2 && git push -f` and then update the PR to go to `master` ?

**Comment by KaelWD** [comment]
yep

File: src/components/VMenu/VMenu.js [hunk 2]
```diff
@@ -190,7 +190,7 @@ export default {
     const data = {
       staticClass: 'v-menu',
       style: {
```
Comment (jacekkarczmarczyk) [inline src/components/VMenu/VMenu.js]: is the style here required at all?

Comment (ebisbe) [reply src/components/VMenu/VMenu.js]: It's just required if the $slots.activator is being used. But it's not printed in the actual markup of the code.

Comment (jacekkarczmarczyk) [reply src/components/VMenu/VMenu.js]: Then I'd add it only if slot is used

Comment (ebisbe) [reply src/components/VMenu/VMenu.js]: Yep, I tried removing the style property and then this:
```
if (this.$slots.activator) {
      data['style']['display'] = this.fullWidth ? 'block' : 'inline-block'
}
```
But tests failed. I searched if I was doing something wrong but couldn't find any explanation. Have any suggestion?

Comment (KaelWD) [reply src/components/VMenu/VMenu.js]: `data.style = {...}`

**Comment by ebisbe** [comment]
done @jacekkarczmarczyk

File: src/components/VMenu/VMenu.js [hunk 3]
```diff
@@ -202,6 +199,10 @@ export default {
       }
     }
 
+    if (this.$slots.activator) {
+      data.style = { display: this.fullWidth ? 'block' : 'inline-block' }
```
Comment (jacekkarczmarczyk) [inline src/components/VMenu/VMenu.js]: One more thing (sorry, I should have noticed it before...) - `.v-menu` has `display: inline-block` by default (https://github.com/ebisbe/vuetify/blob/78417419ac61139170602b05c193f5f2728b327a/src/stylus/components/_menus.styl#L4), so it's enough to set the style only when `this.fullWidth && this.$slots.activator`, and that could be done by using `.v-menu--block` class, so
```js
const data = {
  ...
  class: {'v-menu--block' : this.fullWidth && this.$slots.activator},
  ...
}

Comment (KaelWD) [reply src/components/VMenu/VMenu.js]: Oh, forgot about that. Block will have to be the default if there's no activator.

**Comment by ebisbe** [comment]
Updated the code. I made one test that already had an activator to be fullWidth in order to make sure in the snapshot that we where printing `v-menu--block`

File: src/components/VMenu/VMenu.js [hunk 4]
```diff
@@ -189,9 +189,7 @@ export default {
   render (h) {
     const data = {
       staticClass: 'v-menu',
-      style: {
-        display: this.fullWidth ? 'block' : 'inline-block'
-      },
+      class: { 'v-menu--block': this.fullWidth && this.$slots.activator },
```
Comment (jacekkarczmarczyk) [inline src/components/VMenu/VMenu.js]: as per [Kael's comment](https://github.com/vuetifyjs/vuetify/pull/4577#discussion_r201750297) it should be the other way round - `v-menu` should be `block` by default, and if `!this.fullWidth && this.$slots.activator` it should have `v-menu--inline` class (which needs to be added in .styl)

Comment (ebisbe) [reply src/components/VMenu/VMenu.js]: Ok, perfect. I'm now testing it with the playground and I see that v-menu--block does not exists. I first saw it on the _menus.styl

**Comment by ebisbe** [comment]
updated. I have added a snapshot from the playground.vue you can compare it with https://codepen.io/anon/pen/WzjoGz from the issue itself

**Pull request merged**